### DISCUSSION TITLE
Uncaught TypeError: Cannot read properties of undefined (reading 'on')

### DISCUSSION THREAD

**Discussion by Mr-Shivam-Gupta** [alpinejs/alpine#4148] [1. Help] [answered]
I'm facing trouble while trying to implement a scroll to the top feature and simultaneously run the 'loadMore' method. However, I keep encountering an error: 'Uncaught TypeError: Cannot read properties of undefined (reading 'on')'. I've attempted to use both 'dispatch' and 'emit', but they're also not working and showing the same error. Any suggestions from the community on how to resolve this?

     <div x-data="{ scropTop: 0 }" @scroll="scropTop = $el.scrollTop; if (scropTop <= 0) { window.livewire.on('loadMore'); }">
     //some code ..
     </div>

    protected $listeners = ['loadMore'];
    public function loadMore() : void 
    {
         dd('hit the top');
    }

**Comment by SimoTod** [comment]
The error is on window.livewire which probably doesn't exit. You need to have a look at the livewire documentation to see how you can use it.

**Comment by joshhanley** [comment, marked as answer]
@Mr-Shivam-Gupta questions like this should be asked in the Livewire repo in future.

You don't need to dispatch an event, instead you can just call the method directly using `$wire.loadMore()`.

**Comment by Mr-Shivam-Gupta** [comment]
thankyou so much for the helping me. this is working  $wire.loadMore().
